front/app/containers/Admin/projects/project/analysis/Insights/index.tsx

```diff
@@ -11,19 +11,37 @@ import SummarizeButton from './SummarizeButton';
 import QuestionButton from './QuestionButton';
 import QuestionInput from './QuestionInput';
 import Question from './Question';
+import useAnalysisFilterParams from '../hooks/useAnalysisFilterParams';
+import useInfiniteAnalysisInputs from 'api/analysis_inputs/useInfiniteAnalysisInputs';
 
 const Insights = () => {
   const [isQuestionInputOpen, setIsQuestionInputOpen] = useState(false);
   const { analysisId } = useParams() as { analysisId: string };
   const { data: insights, isLoading } = useAnalysisInsights({
     analysisId,
   });
+  const filters = useAnalysisFilterParams();
+  const { data } = useInfiniteAnalysisInputs({
+    analysisId,
+    queryParams: filters,
+  });
+
+  const inputsCount = data?.pages[0].meta.filtered_count;
 
   return (
     <Box>
       <Box display="flex" gap="4px">
```

Comment:
I removed this flex in my PR, I think it looks better when the buttons are one on top of the other. If we find a way to shorten the text inside, it would look even better.

Reply:
Agreed that this is not the final solution either, it's just a lot better than what we had. Let's ask Rob for some help on this one?